ci-operator/config/openshift/openshift-tests-private/tools/update-cron-entries.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+# Presiquisite:
+# Download https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/tools/generate-cron-entry.sh into $PATH, and make it executable
+
+import yaml
+import sys
+import os
+import subprocess
+from datetime import datetime
+
+def shell(cmd, debug=False, env_=dict(os.environ)):
+    """
+    Args:
+        cmd (string): command to execute
+        debug (bool, optional): print cmd before executing. Defaults to False.
+        env_ (dict, optional): environment variables. Defaults to dict(os.environ).
+
+    Returns:
+        dict[str, str, int]: {"out": stdout, "err": stderr, "rc": return code}
+    """
+    if debug:
+        print("Command: %s" % cmd)
+    result = {"err": "", "out": "", "rc": -1}
+    try:
+        res = subprocess.Popen(cmd, shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE, env=env_)
+        cmd_stdout, cmd_stderr = res.communicate()
+        if sys.version_info[0] < 3:
+            result["err"] = cmd_stderr.strip()
+            result["out"] = cmd_stdout.strip()
+        else:
+            result["err"] = cmd_stderr.strip().decode()
+            result["out"] = cmd_stdout.strip().decode()
+        result["rc"] = res.returncode
+    except Exception as ex:
+        print("Error: %s" % str(ex))
+    return result
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 2:
+        print("Require one parameter: <job-yaml-file>")
+        sys.exit(-1)
+
+    target_file = sys.argv[1]
+    print("Updating %s" %(target_file))
+
+    with open(target_file, 'r') as file:
+        all_data = yaml.safe_load(file)
+    file.close()
+
+    all_tests_list = all_data['tests']
+    new_tests_list = []
+    for test in all_tests_list:
+        test_name = test['as']
+        print("updating test job - %s" %(test_name))
+        cron_output = shell("generate-cron-entry.sh %s %s" %(test_name, target_file))["out"]
+        cron_entry = cron_output.split("cron:")[-1].strip()
+        print(cron_entry)
+        test['cron'] = cron_entry
+        new_tests_list.append(test)
+    
+    all_data['tests'] = new_tests_list
+    
+    backup_file = "%s.%s" %(target_file, datetime.now().strftime("%Y%m%d%H%M%S"))
```

Comment:
do we really need this backup file? b/c everyone is using git.

Reply:
I cam turn it off, but actually i leave it there to give users a chance to compare for safety.

Reply:
I did some update to introduce `--backup` option, users can disable it via `--backup=no`, also the tool can support to edit multiple yaml files in one line.